docs(react-native): UI component - CallContentView (#803)
GetStream/stream-video-js: packages/react-native-sdk/docusaurus/docs/reactnative/04-ui-components/call/call-content-view.mdx

title: CallContentView
---

> CallContentView is one of the main components for a Call. This is responsible for listing all the participant's video stream in a vertical/horizontal list and play the audio streams for those participants.

Reviewer: ```suggestion
CallContentView is one of the main components for a Call. This is responsible for listing all the participant's video streams in a vertical/horizontal list and playing the audio streams for those participants.
```

![Preview of the CallContentView component.](../../assets/04-ui-components/call-content-view-grid.png)

> The CallContentView has two modes(`grid` or `spotlight`). The grid mode simply lists the participants vertically one after the other, while the spotlight mode is responsible to show the screen sharing view, a single dominant speaker, or a pinned participant in an elongated view, while other participants being in an horizontal view.

Reviewer: ```suggestion
The CallContentView has two modes(`grid` or `spotlight`). The grid mode lists the participants vertically, one after the other. In contrast, the spotlight mode shows the screen sharing view, or a single dominant speaker, or a pinned participant in an elongated view, while other participants are in a horizontal view.
```

![Preview of the CallContentView component in spotlight mode](../../assets/04-ui-components/call-content-view-spotlight.png)

## General usage

The `CallContentView` can be used to display the video stream of all the participants and play their audio stream.

```tsx
import {
  Call,
  CallContentView,
  StreamCall,
} from '@stream-io/video-react-native-sdk';

const VideoCallUI = () => {

>   let call: Call;

Reviewer: ```suggestion
  let call: Call;
  // your logic to create or getting an existing call 
```

  return (
    <StreamCall call={call}>
      // highlight-next-line
      <CallContentView />
    </StreamCall>
  );
};
```

## Props

### `mode`

| Type                  | Default Value |
| --------------------- | ------------- |
| `grid` \| `spotlight` | `grid`        |

> This is used to switch the list between the grid and the spotlight mode.

Reviewer: ```suggestion
This switches the list between the grid and the spotlight mode.
```
